### PR TITLE
OADP-2130: Reworking the Installation and Features section

### PR DESCRIPTION
**Pull request by anarnold97**
[OADP-2130](https://issues.redhat.com/browse/OADP-2130)

Re-working the following sections:
* Installing OADP
* OADP features and plugins
* Configuring OADP
* Uninstalling OADP

*Installation

Version(s):

Issue:

https://issues.redhat.com/browse/OADP-2130

Link to docs preview:

* [Installing OADP](https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.html)
* [OADP features and plugins](https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/features-and-plugins/oadp-features-plugins.html)
* [Configuring OADP](https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-intro.html)
* [Uninstalling OADP](https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/uninstalling-oadp.html)

QE review:
- [ ] QE has approved this change.

Additional information:

Apologies, but the only way I can realistically manage the reviews from the OCP team is to split this up

### CODE REVIEW

**Comment by anarnold97** [comment]
@weshayutin & @akarol - please can you arrange a review of the structure changes for these 4 sections
Apologies, I cannot manage to get all the changes through OCP review at once, so need to break it up into parts
Thanks

**Comment by stillalearner** [comment]
@anarnold97 

Regarding the Configuring OADP section:

There are many parts which are duplicated for every provider, but those parts have nothing to do with the specific providers.

For example:

1. https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.html#oadp-setting-resource-limits-and-requests_installing-oadp-gcp
2. https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.html#oadp-self-signed-certificate_installing-oadp-gcp
3. https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.html#oadp-enabling-csi-dpa_installing-oadp-gcp
4. https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.html#oadp-about-backup-snapshot-locations_installing-oadp-gcp
5. https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.html#oadp-installing-dpa_installing-oadp-gcp

All these parts specify the common configurations for all the providers and how any DPA can be configured. 
So these parts should be added to an introduction page, which first explains the parts of any DPA, like backup locations, secrets, etc. (4th point).
Then, the installation part of DPA (5th point), keeping the secret name independent of the provider on the first page.
After that, other configurations  of DPA like resource allocations / CSI/ self-signed CA certificates (1, 2, 3 points).


Then, on the cloud provider specific pages, only the default secret name for that provider and the configuration steps would be needed.
For example : for GCP, only this:
https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp.html#migration-configuring-gcp_installing-oadp-gcp

Followed by secret creation steps and example DPA CR for the provider.

**Comment by anarnold97** [comment]
@stillalearner - Thanks very much. I see what you mean. 
I will make the change this evening and let us see if it works.

**Comment by anarnold97** [comment]
@stillalearner - this is actually more difficult than we thought. 
When I looking at the modules, they are built `if` statements and there are small changes between some of the 
See `oadp-secrets-for-different-credentials.adoc`
You will see this is using:
```
ifdef::installing-oadp-aws[]
```
```
ifdef::installing-oadp-azure,installing-oadp-gcp,installing-oadp-ocs,installing-oadp-mcg[]
``` 
So will look very odd if we start splitting it into single pages, as then the context will not work.

**Comment by stillalearner** [comment]
@anarnold97 went through the changes, now it is looking better,  we just need to have few more changes.

First page -> Introduction to configuring oadp looks good.

Second page -> Creating backup and snapshot locations

This page also looks good, but after the definitions of backup, snapshot locations, and secret, 

The creation of default secret part should also be added to this page only as it is common to all.

https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.html#oadp-creating-default-secret_installing-oadp-aws


After that, on the specific cloud provider pages, we can keep the "Create secrets for different credentials" as it is.


Then, finally the 1st , 2nd, and 3rd point sections mentioned in the last comment can be combined as a single page, mentioning "additional DPA configurations".

Please let me know if there are any issues in this.

File: backup_and_restore/application_backup_and_restore/features-and-plugins/oadp-features.adoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+toc::[]
+
+OpenShift API for Data Protection (OADP) supports the following features:
+
+[id="oadp-features-backup"]
+== Backup
+
+You can use OADP to back up all applications on the OpenShift Platform, or you can filter the resources by type, namespace, or label.
++
+OADP backs up Kubernetes objects and internal images by saving them as an archive file on object storage. OADP backs up persistent volumes (PVs) by creating snapshots with the native cloud snapshot API or with the Container Storage Interface (CSI).
```
Comment (amastbau) [inline backup_and_restore/application_backup_and_restore/features-and-plugins/oadp-features.adoc:20]: maybe: them as an archive file on an/the object storage, might be worng

File: backup_and_restore/application_backup_and_restore/installing/installing-oadp-intro.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
+:_content-type: ASSEMBLY
+[id="installing-oadp-intro"]
+= Introduction to configuring the OpenShift API for Data Protection
+include::_attributes/common-attributes.adoc[]
+:context: installing-oadp-cloud
+
+toc::[]
+
+You install the {oadp-first} with your backup storage provider by installing the OADP Operator.
```
Comment (amastbau) [inline backup_and_restore/application_backup_and_restore/installing/installing-oadp-intro.adoc:9]: the preview is not available,
just to clarify:
oadp-first is a doc parameter, correct?

File: backup_and_restore/application_backup_and_restore/features-and-plugins/oadp-features-plugins.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+include::modules/oadp-features.adoc[leveloffset=+1]
+include::modules/oadp-plugins.adoc[leveloffset=+1]
+include::modules/oadp-configuring-velero-plugins.adoc[leveloffset=+1]
+
+[id="oadp-support-for-ibm-power-and-ibm-z"]
+== OADP support for IBM Power and {ibmzProductName}
+
+OpenShift API for Data Protection (OADP) is platform neutral. The information that follows relates only to IBM Power and to {ibmzProductName}.
+
+OADP 1.1.0 was tested successfully against {product-title} 4.11 for both IBM Power and {ibmzProductName}. The sections that follow give testing and support information for OADP 1.1.0 in terms of backup locations for these systems.
```
Comment (amastbau) [inline backup_and_restore/application_backup_and_restore/features-and-plugins/oadp-features-plugins.adoc:25]: Is this 1.2 or 1.1 documentation?
am i missing somthing?

File: backup_and_restore/application_backup_and_restore/installing/installing-oadp-intro.adoc (hunk 2 above)
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/installing/installing-oadp-intro.adoc:9]: **Preview -** 
I am not sure why you cannot see the preview:

* [Introduction to installing OADP ](https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.html)
* [OADP features and plugins](https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/features-and-plugins/oadp-features-plugins.html)

**_attributes/common-attributes.adoc =>>** 
Yes added
*:oadp-first: OpenShift API for Data Protection (OADP)* - I need to push this through the docs and replace the wording as it should have done when the docs were first done.

File: backup_and_restore/application_backup_and_restore/features-and-plugins/oadp-features-plugins.adoc (hunk 3 above)
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/features-and-plugins/oadp-features-plugins.adoc:25]: The docs cover 1.2 and 1.1. 

But I get your point, I need to create a *doc improvements* ticket to update the docs for this section for 1.2

File: backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc [hunk 4]
```diff
@@ -1,12 +1,21 @@
...
 :context: about-installing-oadp
+:_content-type: CONCEPT
+// as it is not really a prerequisites list
+// should we keep that same pattern
+// to resolve the duplication
 
 toc::[]
 
-As a cluster administrator, you install the OpenShift API for Data Protection (OADP) by installing the OADP Operator. The OADP Operator installs link:https://{velero-domain}/docs/v{velero-version}/[Velero {velero-version}].
+As a cluster administrator, you use the OpenShift API for Data Protection (OADP) by installing the OADP Operator. The OADP Operator installs link:https://{velero-domain}/docs/v{velero-version}/[Velero {velero-version}], and OpenShift plugins for Velero to use, for backup and restore operations.
```
Comment (amastbau) [inline backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc:13]: i think that in the preview I see velero 1.9
but there might be a comment for that already

Comment (amastbau) [reply backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc:13]: (I was looking at the wrong place, I see the previews now, sat.)

**Comment by amastbau** [comment]
Generic comment

In a lot of places I have noticed
"how many minutes to wait"
while it can be another time length then minutes.
This is en example from the crd reouse we should be the source of true i think
VERSION:  oadp.openshift.io/v1alpha1

FIELD:    resourceTimeout <string>

DESCRIPTION:
     resourceTimeout defines how long to wait for several Velero resources
     before timeout occurs, such as Velero CRD availability, volumeSnapshot
     deletion, and repo availability. Default is 10m

File: backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+[IMPORTANT]
+====
+Restic does not support backing up `hostPath` volumes. For more information, see link:https://{velero-domain}/docs/v{velero-version}/restic/#limitations[additional Restic limitations].
+====
+
+.Prerequisites
+
+* You must install the OpenShift API for Data Protection (OADP) Operator.
+* You must not disable the default Restic installation by setting `spec.configuration.restic.enable` to `false` in the `DataProtectionApplication` CR.
+* The `DataProtectionApplication` CR must be in a `Ready` state.
```
Comment (amastbau) [inline backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc:34]: Is this considered a Ready State?
status:
    conditions:
    - lastTransitionTime: "2023-07-23T18:17:29Z"
      message: Reconcile complete
      reason: Complete
      status: "True"
      type: Reconciled
   maybe i missing something

**Comment by duduvaa** [comment]
https://62396--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.html

For OADP 1.2.0 - if datamover "enable:  true" , Restic must be "enable: false"

**Comment by duduvaa** [comment]
https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-api.html

Table10 - OADP 1.2.0:
if datamover "enable: true" , need to add 'vsm' to velero defaultPlugins list.
if datamover "enable: false" or not exists, need to verify 'vsm' is not exists in velero defaultPlugins list

File: backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc (hunk 5 above)
Comment (anarnold97) [reply backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc:34]: @amastbau - please can you help me with what this should be? thanks

**Comment by anarnold97** [comment]
> https://62396--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.html
> 
> For OADP 1.2.0 - if datamover "enable: true" , Restic must be "enable: false"

I think you might be looking at the wrong section - 

> https://62428--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-api.html
> 
> Table10 - OADP 1.2.0: if datamover "enable: true" , need to add 'vsm' to velero defaultPlugins list. if datamover "enable: false" or not exists, need to verify 'vsm' is not exists in velero defaultPlugins list

Hi @duduvaa - I am not sure you are looking at the right section. This is mainly installation, and ,features and plugins rework sections rework. I think you need this PR - https://github.com/openshift/openshift-docs/pull/62431

I hope you do cos otherwise I have major problems.

File: backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc (hunk 5 above)
Comment (amastbau) [reply backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc:34]: I would suggest in The `DataProtectionApplication` CR must be with the status `Condition: Reconciled`
@anarnold97 
I wonder how this status condition is referred to in other product documentation
This is a comment status for a CR.

**Pull request closed without merging**